src/main/java/org/jabref/gui/search/GlobalSearchBar.java

```diff
@@ -231,10 +238,29 @@ public GlobalSearchBar(LibraryTabContainer tabContainer,
     }
 
     private void initSearchModifierButtons() {
-        fulltextButton.setSelected(searchPreferences.isFulltext());
         fulltextButton.setTooltip(new Tooltip(Localization.lang("Fulltext search")));
         initSearchModifierButton(fulltextButton);
+
+        EasyBind.subscribe(filePreferences.fulltextIndexLinkedFilesProperty(), enabled -> {
+            if (!enabled) {
+                fulltextButton.setSelected(false);
+            } else if (searchPreferences.isFulltext()) {
+                fulltextButton.setSelected(true);
+            }
+        });
+
         fulltextButton.selectedProperty().addListener((obs, oldVal, newVal) -> {
+            if (!filePreferences.shouldFulltextIndexLinkedFiles() && newVal) {
+                boolean enableFulltextSearch = dialogService.showConfirmationDialogAndWait(Localization.lang("Fulltext search"), Localization.lang("Fulltext search requires the setting 'Automatically index all linked files for fulltext search' to be enabled. Do you want to enable indexing now?"), Localization.lang("Enable indexing"), Localization.lang("Keep disabled"));
+
+                LibraryTab libraryTab = tabContainer.getCurrentLibraryTab();
+                if (libraryTab != null && enableFulltextSearch) {
+                    filePreferences.setFulltextIndexLinkedFiles(true);
+                }
+                if (!enableFulltextSearch) {
+                    fulltextButton.setSelected(false);
+                }
+            }
             searchPreferences.setSearchFlag(SearchFlags.FULLTEXT, newVal);
```

Comment:
Store the correct value of the flag, it will fix the issue.
```suggestion
                    filePreferences.setFulltextIndexLinkedFiles(true);
                    searchPreferences.setSearchFlag(SearchFlags.FULLTEXT, true);
                }
                if (!enableFulltextSearch) {
                    fulltextButton.setSelected(false);
                    searchPreferences.setSearchFlag(SearchFlags.FULLTEXT, false);
                }
            } else {
                searchPreferences.setSearchFlag(SearchFlags.FULLTEXT, newVal);
            }
```

Reply:
Thanks!